### PR TITLE
use async prometheus

### PR DESCRIPTION
**Pull request by KPostOffice**
## Related Issues and Dependencies

https://github.com/thoth-station/investigator/issues/274

## This introduces a breaking change

- [ ] Yes
- [x] No

## This Pull Request implements

Use prometheus async library for metrics

### CODE REVIEW

File: investigator/investigator/advise_justification/investigate_advise_justification.py [hunk 1]
```diff
@@ -24,14 +24,15 @@
...
+from prometheus_async.aio import track_inprogress, count_exceptions
 
 _LOGGER = logging.getLogger(__name__)
 
 DEPLOYMENT_NAME = os.environ["THOTH_DEPLOYMENT_NAME"]
 
 
-@advise_justification_exceptions.count_exceptions()
-@advise_justification_in_progress.track_inprogress()
+@count_exceptions(advise_justification_exceptions)
```
Comment (pacospace) [inline investigator/investigator/advise_justification/investigate_advise_justification.py:34]: we do not need one metric file per agent in this case then, we can define one metric common to all and use labels, WDYT?

Comment (KPostOffice) [reply investigator/investigator/advise_justification/investigate_advise_justification.py:34]: That's what we do. The individual metrics files just set the labels

Comment (pacospace) [reply investigator/investigator/advise_justification/investigate_advise_justification.py:34]: I thought we do not need those files anymore now as you can assign it directly in the main file. then LGTM

**Pull request merged**